starter_code/public/javascript/APIHandler.js

```diff
@@ -4,22 +4,145 @@ class APIHandler {
   }
 
   getFullList () {
+    $.ajax({
+    method:"GET",
+    url: "https://ih-crud-api.herokuapp.com/characters"
+  })
+  .then((apiResult)=>{
+    console.log("Success!");
+    console.log(apiResult);
+    const charArray=apiResult;
+    $(".character-loop").empty();
+    charArray.forEach((oneCharacter)=>{
+      $(".character-loop").append(
+        `
+        <div class="characters-container">
+          <div class="character-info">
+            <div class="name">Name: ${oneCharacter.name}</div>
+            <div class="occupation">Occupation: ${oneCharacter.occupation}</div>
+            <div class="debt">Debt: ${oneCharacter.debt}</div>
+            <div class="weapon">Weapon: ${oneCharacter.weapon}</div>
+            <div class="id">ID: ${oneCharacter.id}</div>
+          </div>
+        </div>
+        `
+      );
+    });
+  })//end then
+  .catch((err)=>{
+    console.log(err);
+    console.log("ERROR!");
+  });//end catch
+}//getFullList()
 
-  }
 
-  getOneRegister () {
+getOneRegister (oneId) {
 
-  }
+  $.ajax({
+    method:"GET",
+    url:`https://ih-crud-api.herokuapp.com/characters/${oneId}`
+  })
+  .then((apiResult)=>{
+    $(".character-loop").empty();
+    console.log("SUCCESS!");
+    console.log(apiResult);
+    $(".character-loop").append(
+      `
+      <div class="characters-container">
+        <div class="character-info">
+          <div class="name">${apiResult.name}</div>
+          <div class="occupation">${apiResult.occupation}</div>
+          <div class="debt">${apiResult.debt}</div>
+          <div class="weapon">${apiResult.weapon}</div>
+        </div>
+      </div>
+      `
+    );
+  })
+  .catch((err)=>{
+    console.log("ERROR!");
+    console.log(err);
+  });
+
+}
 
-  createOneRegister () {
+  createOneRegister (newName,newWeapon,newDebt,newJob) {
+    $.ajax({
+    method:"POST",
+    url: "https://ih-crud-api.herokuapp.com/characters",
+    //data to send through in the form body
+    data: {
+      name:newName,
+      weapon:newWeapon,
+      debt: newDebt,
+      occupation:newJob
+    }
+  })
+  .then((apiResult)=>{
+    console.log("Success!");
+    console.log(apiResult);
+    $(".create-new-feedback").append(`
+      <li>
+        Added <b> ${apiResult.name}</b>
+        (id ${apiResult.id})
+       </li>
+      `);
+  })
+  .catch((err)=>{
+    console.log(err);
+    console.log("ERROR!");
+  });
 
   }
 
-  updateOneRegister () {
+  updateOneRegister (editedId, editedName, editedJob, editedDebt,editedWeapon) {
+    $.ajax({
+    method:"PATCH",
```

Comment:
Nice experimentaion with PATCH. (It does the same as PUT but it's nice to see it in action).